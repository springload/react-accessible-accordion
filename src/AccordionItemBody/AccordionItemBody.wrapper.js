// @flow

import React, { Component } from 'react';
import type { ElementProps } from 'react';

import { Subscribe } from 'unstated';
import AccordionContainer from '../AccordionContainer/AccordionContainer';
import ItemContainer from '../ItemContainer/ItemContainer';
import AccordionItemBody from './AccordionItemBody';

type AccordionItemBodyWrapperProps = ElementProps<'div'> & {
    hideBodyClassName: string,
};

class AccordionItemBodyWrapper extends Component<
    AccordionItemBodyWrapperProps,
> {
    static defaultProps = {
        className: 'accordion__body',
        hideBodyClassName: 'accordion__body--hidden',
    };

    renderItemBody = (
        accordionStore: AccordionContainer,
        itemStore: ItemContainer,
    ) => {
        const { uuid } = itemStore.state;
        const { items, accordion } = accordionStore.state;
        const item = items.find(stateItem => stateItem.uuid === uuid);
        return (
            <AccordionItemBody
                {...this.props}
                {...item}
                accordion={accordion}
            />
        );
    };

    render() {
        return (
            <Subscribe to={[AccordionContainer, ItemContainer]}>
                {this.renderItemBody}
            </Subscribe>
        );
    }
}

export default AccordionItemBodyWrapper;
