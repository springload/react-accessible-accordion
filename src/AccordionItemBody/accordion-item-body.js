import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

const defaultProps = {
    id: '',
    expanded: false,
    className: 'accordion__body',
    hideBodyClassName: 'accordion__body--hidden',
    role: '',
};

const propTypes = {
    id: PropTypes.string,
    expanded: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element),
    ]).isRequired,
    className: PropTypes.string,
    hideBodyClassName: PropTypes.string,
    role: PropTypes.string,
};

const AccordionItemBody = ({ id, expanded, children, className, hideBodyClassName, role }) => {
    const bodyClass = classNames(
        className,
        {
            [hideBodyClassName]: !expanded,
        },
    );
    const ariaHidden = !expanded;
    return (
        <div
            id={id}
            className={bodyClass}
            aria-hidden={ariaHidden}
            aria-labelledby={id.replace('accordion__body-', 'accordion__title-')}
            role={role}
        >
            {children}
        </div>
    );
};

AccordionItemBody.propTypes = propTypes;
AccordionItemBody.defaultProps = defaultProps;
// We need this to be able to assign correct params to element.
// Minifiers modify component name
AccordionItemBody.accordionElementName = 'AccordionItemBody';

export default AccordionItemBody;
