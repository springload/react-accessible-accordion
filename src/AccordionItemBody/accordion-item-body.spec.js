import React from 'react';
import renderer from 'react-test-renderer';

import AccordionItemBody from './accordion-item-body';

describe('AccordionItemBody', () => {
    it('renders correctly with min params', () => {
        const tree = renderer.create(
            <AccordionItemBody>
                <div>Fake body</div>
            </AccordionItemBody>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with different className', () => {
        const tree = renderer.create(
            <AccordionItemBody className="testCSSClass">
                <div>Fake body</div>
            </AccordionItemBody>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with prefixClass', () => {
        const tree = renderer.create(
            <AccordionItemBody hideBodyClassName="testCSSClass--hidden">
                <div>Fake body</div>
            </AccordionItemBody>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with expanded', () => {
        const tree = renderer.create(
            <AccordionItemBody expanded={true}>
                <div>Fake body</div>
            </AccordionItemBody>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with an id (aria-labelledBy filled)', () => {
        const tree = renderer.create(
            <AccordionItemBody id="accordion__body-HASHID" expanded={true}>
                <div>Fake body</div>
            </AccordionItemBody>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
