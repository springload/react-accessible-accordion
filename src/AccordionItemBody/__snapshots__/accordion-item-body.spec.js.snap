// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemBody renders correctly with an id (aria-labelledBy filled) 1`] = `
<div
  aria-hidden={false}
  aria-labelledby="accordion__title-HASHID"
  className="accordion__body"
  id="accordion__body-HASHID"
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with different className 1`] = `
<div
  aria-hidden={true}
  aria-labelledby=""
  className="testCSSClass accordion__body--hidden"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with expanded 1`] = `
<div
  aria-hidden={false}
  aria-labelledby=""
  className="accordion__body"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with min params 1`] = `
<div
  aria-hidden={true}
  aria-labelledby=""
  className="accordion__body accordion__body--hidden"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with prefixClass 1`] = `
<div
  aria-hidden={true}
  aria-labelledby=""
  className="accordion__body testCSSClass--hidden"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;
